### ISSUE TITLE
Allow shutdown of underlying Socket

### ISSUE THREAD

**Issue by matthid** [elmish/debugger#36]
### Description

We use the debugger in the integration-test suite (to test various stuff) and in order to make it work properly we need to shutdown the underlying socket.
To do this we need:
 - https://github.com/zalmoxisus/remotedev/pull/17
 - Some changes in this Repository to use this new `destroy` function.

While technically this requires https://github.com/elmish/elmish/issues/162
We have implemented it by "abusing" the `init` function in the following way: If `init` is called with a parameter of `null` we assume this is a program shutdown, which is good enough for our testing scenario.

This issue is a remainder and a starting point for discussion.
Is there interest in bringing these changes to this repository, maybe even depending on my fork of `remotedev`?

I spoke about this issue in the past [here](https://github.com/elmish/debugger/pull/21#issuecomment-466520210) and [here](https://github.com/elmish/debugger/pull/22#issuecomment-466527648)

**Comment by et1975** [comment]
Yes to both, I think. I haven't heard anything from Tim, but I guess we can move forward with the shutdown `alpha` release (thanks @MangelMaxime for catching that nonsense :)).

**Comment by matthid** [comment]
So you mean opening a PR for custom fork (or maybe even just bundling it) and the init abuse? :D

**Comment by et1975** [comment]
I mean let's implement the termination mechanism, as introduced in elmish v4 and the dependency on your fork seems ok to me. We don't include or reference JS dependencies, so it's just a matter of documenting it for the next major release, at least until the entire elmish stack is on Femto.

**Comment by et1975** [comment]
@matthid the `master` is ready to take these changes now.
